modules/ecs-service/main.tf

```diff
@@ -0,0 +1,696 @@
+
+  # Our applications are designed to have quick response times
+  # therefore we can afford using a shorter timeout.
+  # This provides faster draining of unhealthy or deregistering
+  # of tasks from the target group. Example being a deployment
+  # where the tasks are being replaced with new tasks. Default
+  # would take 5 minutes to deregister a task.
+  deregistration_delay = 5
+
+  health_check {
+    # This is the default health check configuration for the target group.
+    # This would mean that a task could be considered healthy in 2 * (10 + 5) = 30 seconds.
+    # or the task could be considered unhealthy in 3 * (5 + 10) = 45 seconds.
+    healthy_threshold   = 2
+    unhealthy_threshold = 3
+    timeout             = 5
+    interval            = 10
+    path                = "/"
+    matcher             = "200"
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+
+# ------------------------------------------------------------
+
+# THE FOLLOWING SECTION IS USED TO CONFIGURE
+
+# SCHEDULED TASKS. WHEN A SCHEDULED TASK IS
+
+# CONFIGURED, THE ECS SERVICE RESOURCE SHOULD
+
+# NOT BE CREATED.
+
+# ------------------------------------------------------------
+
+
+# -------------------------------------------
+# CREATE THE ECS TARGET FOR THE RULE
+# -------------------------------------------
+
+resource "aws_cloudwatch_event_target" "scheduled" {
+  count = var.create_scheduled_task ? 1 : 0
+
+  target_id = "${var.ecs_service_name}-scheduled"
+  arn       = data.aws_ecs_cluster.cluster.arn
+  rule      = aws_cloudwatch_event_rule.scheduled[0].name
+  role_arn  = aws_iam_role.scheduled[0].arn
+
+  ecs_target {
+    task_count          = var.desired_number_of_tasks
+    task_definition_arn = local.task_definition
+    launch_type         = "FARGATE"
+    network_configuration {
+      subnets          = var.scheduled_task_subnet_ids
+      security_groups  = var.scheduled_task_security_group_ids
+      assign_public_ip = var.scheduled_task_assign_public_ip
+    }
+  }
+}
+
+
+# -------------------------------------------
+# CREATE THE RULE TO TRIGGER THE ECS TASK
+# -------------------------------------------
+
+resource "aws_cloudwatch_event_rule" "scheduled" {
+  count = var.create_scheduled_task ? 1 : 0
+
+  name        = "${var.ecs_service_name}-rule"
+  description = "Trigger the ECS task by schedule or event."
+
+  event_pattern       = var.scheduled_task_event_pattern != null ? jsonencode(var.scheduled_task_event_pattern) : null
+  schedule_expression = var.scheduled_task_cron_expression != "" ? var.scheduled_task_cron_expression : null
+
+  depends_on = [aws_ecs_task_definition.task]
+}
+
+
+# -------------------------------------------
+# CREATE THE ECS IAM ROLE FOR THE EVENT
+# -------------------------------------------
+
+resource "aws_iam_role" "scheduled" {
+  count = var.create_scheduled_task ? 1 : 0
+
+  name_prefix = "${var.ecs_service_name}-scheduled"
+  assume_role_policy = jsonencode({
```

Comment:
You already know what Imma say: aws_iam_policy_document. Another use case where if you used the terraform resource, you could just use `assume_role_policy = aws_iam_policy_document.assume_role`